File: javascript/p6.js
#!/usr/bin/env node

let limit = 100
if (process.argv.length > 2){
  limit = process.argv[2];
}
let sum = limit * (limit + 1) / 2;
let sum_sq = (2 * limit + 1) * (limit + 1) * limit / 6;
let diff = Math.pow(sum, 2) - sum_sq;
console.log(diff);

File: perl/p6.pl
#!/usr/bin/perl

$limit = 100;
if (scalar(@ARGV) > 0){
  $limit = $ARGV[0];
}
$sum = $limit * ($limit + 1) / 2;
$sum_sq = (2 * $limit + 1) * ($limit + 1) * $limit / 6;
$diff = ($sum ** 2) - $sum_sq;
printf "%d\n", $diff;
